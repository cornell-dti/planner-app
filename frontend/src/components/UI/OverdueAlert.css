.OverdueAlertRelative {
  position: absolute;
  width: 2em;
  height: 2em;
  background-color: #D0021B;
  line-height: 2em;
  text-align: center;
  font-weight: bold;
  font-size: 1em;

  /* Proposed Experimental Change */
  top: -0.3em;
  right: -0.3em;
  /*border-bottom-left-radius: 1em;*/
}

.OverdueAlertAbsolute {
  position: absolute;
  z-index: 1;
  width: 1em;
  height: 1em;
  background-color: #D0021B;
  line-height: 1em;
  text-align: center;
  font-weight: bold;
  font-size: 0.5em;
}
