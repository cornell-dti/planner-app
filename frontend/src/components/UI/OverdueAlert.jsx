// @flow strict

import React from 'react';
import type { Node } from 'react';
import styles from './OverdueAlert.css';

type Props = {|
  +absolutePosition: ?{| +top: number; +left: number; |};
|};

export default function OverdueAlert({ absolutePosition }: Props): Node {
  if (absolutePosition == null) {
    return <div className={styles.OverdueAlertRelative}>!</div>;
  }
  const { top, left } = absolutePosition;
  const style = { top: `${top}px`, left: `${left}px` };
  return <div className={styles.OverdueAlertAbsolute} style={style}>!</div>;
}

// eslint-disable-next-line react/default-props-match-prop-types
OverdueAlert.defaultProps = { absolutePosition: null };
