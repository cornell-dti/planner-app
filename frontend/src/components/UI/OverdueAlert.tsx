import React, { ReactElement } from 'react';
import styles from './OverdueAlert.module.css';

type Props = { readonly target: 'task-card' | 'future-view-task' };

/**
 * The overdue alert that will be displayed on the top-right corner of a div.
 */
export default function OverdueAlert({ target }: Props): ReactElement {
  const className = target === 'task-card'
    ? styles.OverdueAlertInTaskCard
    : styles.OverdueAlertInFutureView;
  return <div className={className}>!</div>;
}
