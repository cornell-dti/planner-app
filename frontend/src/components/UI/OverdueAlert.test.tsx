import React from 'react';
import renderer from 'react-test-renderer';
import OverdueAlert from './OverdueAlert';

it('OverdueAlert in future-view-task matches snapshot.', () => {
  const tree = renderer.create(<OverdueAlert target="future-view-task" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('OverdueAlert in task-card matches snapshot.', () => {
  const tree = renderer.create(<OverdueAlert target="task-card" />).toJSON();
  expect(tree).toMatchSnapshot();
});
