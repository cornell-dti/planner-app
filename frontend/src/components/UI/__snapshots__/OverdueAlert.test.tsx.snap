// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverdueAlert in future-view-task matches snapshot. 1`] = `
<div
  className="OverdueAlertInFutureView"
>
  !
</div>
`;

exports[`OverdueAlert in task-card matches snapshot. 1`] = `
<div
  className="OverdueAlertInTaskCard"
>
  !
</div>
`;
